src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ReferralService.kt

```diff
@@ -141,16 +142,34 @@ class ReferralService(
 
     /*
      * This is a temporary solution until a robust asynchronous link is created between Interventions
-     * and Delius. Once the asynchronous link is implemented, this feature can be turned off, and instead
-     * Delius will be notified via the normal asynchronous route
+     * and Delius/ARN. Once the asynchronous link is implemented, this feature can be turned off, and instead
+     * Delius/ARN will be notified via the normal asynchronous route
      */
     communityAPIReferralService.send(referral)
+    submitAdditionalRiskInformation(referral, user)
```

Comment:
Sorry, asking the transactional question 😅 

What should our behaviour be when the latter (risk info) does not succeed and we cannot persist the sent referral?

Reply:
yeh i have thought about this, and right now we are pretty stuck if that happens. 

answer about community api idempotency is 

"Then we will end up getting two NSIs and then when you progress the referral it will complain as it finds multiple"

Reply:
Can we perhaps defer it after persistence? Make it a two-stage process?

1. community-api
2. save
3. event
4. risk
5. save

?

Reply:
Based on: observing state will tell us if the persistence was successful or not, so we can always retry this on sent referrals that still have the risk info

Reply:
Ok I muddled this a bit. I meant making the transaction successful just after points 1-3.

But it's ok -- our recovery runbook for this is: delete the NSI and let the user try again

Reply:
Added it under https://dsdmoj.atlassian.net/wiki/spaces/IC/pages/3386179786/Recovery+scenarios#ARN-call-fails-and-the-referral-cannot-be-sent